Upgrade django

hluk opened the pull request:
---

Bumping Django to 1.11.

lubomir commented:
When running the tests, I get this warning:

```
/home/lsedlar/.virtualenvs/pdc-django-1.11/lib/python2.7/site-packages/django/db/models/fields/__init__.py:1451: RuntimeWarning: DateTimeField Changeset.committed_on received a naive datetime (2015-02-03 02:55:18) while time zone support is active.
  RuntimeWarning)
```

The script to check if migrations are created is crashing in travis, but somehow it does not affect anything.

hluk commented:
I fixed the "missing time zone" warning.

I'll try to do something with "verify-migrations.sh" on Travis.

hluk commented:
Migrations fixed! The fix is not to add image formats and types from productmd when PDC server is started (done only after migrations are applied). This doesn't make much sense to me.

lubomir commented:
The reasoning for inserting on start up is when new values are added in productmd, a restart would be enough to pick them up and populate the database.

lubomir commented:
Running `python manage.py migrate` even with up-to-date database is however enough to emit the post migrate signal, so this is not really an issue.

xiangge commented on pdc/apps/package/models.py:
```diff
@@ -366,13 +364,6 @@ def sync_image_formats_and_types():
-        return
-    sync_image_formats_and_types()
-
-
```
After the first time migration, when we restart the service for some reasons,  don't we need this function?
What error you met? Not sure if the productmd's data will update sometimes.

hluk replied:
Basically if migrations didn't run yet (db tables don't exist) this would fail.

I would need to change the condition above for checking parameters - but that can break easily in future or with some other parameters.

If productmd is updated while installing new version of PDC, this is not an issue because migrations should be executed and then `sync_image_formats_and_types()` is run.

Not sure if only productmd can be updated without running PDC ansible scripts. 

@lubomir Do you know whether PDC ansible scripts were used last time productmd was updated?

lubomir replied:
I'm not sure about production, but at least stage was updated manually. I don't see it as a big concern anyway: we generally know if there are any changes, and could ask for the migrations to be run as well.

xiangge replied:
If there is no migration at the beginning, I think many errors will raise. So maybe this situation should ignore. 
Not sure how productmd updates, but I think if we keep this trigger here when there is an update request, just need to restart the service (apache).
That's my thinking, but I am also fine with removing these codes as we can do a manually migration.

xiangge commented on pdc/apps/changeset/filters.py:
```diff
@@ -23,7 +23,7 @@ class ChangesetFilterSet(django_filters.FilterSet):
                                                 widget=widgets.DateTimeInput)
-    comment = django_filters.CharFilter(name="comment", lookup_type="contains")
+    comment = django_filters.CharFilter(name="comment", lookup_expr="contains")
 
```
Why we need change the lookup_type to lookup_expr, is this related to upgrade django? 
Is that because this lookup_type not work any more?

xiangge commented on pdc/apps/release/serializers.py:
```diff
@@ -151,17 +151,6 @@ def update(self, instance, validated_data):
-                        if hasattr(obj, attr):
-                            setattr(obj, attr, None)
-                    except ValueError:
-                        pass
```
Do you mean in 1.11 the put option have already erased the optional fields?
If so, I hope we can create an issue to track this change and let QE guys to test it.

hluk replied:
I have no idea why it's not needed now but the tests pass (and there are specific tests for this behavior).

hluk replied:
Just tested the behavior:
```bash
> pdc_client -r releases/acbu-8271 -x PUT -d '{"release_type": "ga", "short": "acbu", "version": "8271", "name": "adcjysacbu", "allow_buildroot_push": "true"}'
{
    ...
    "allow_buildroot_push": true, 
    ...
}

> pdc_client -r releases/acbu-8271 -x PUT -d '{"release_type": "ga", "short": "acbu", "version": "8271", "name": "adcjysacbu"}'                                
{
    ...
    "allow_buildroot_push": false, 
    ...
}
```

xiangge replied:
Yes, I saw the tests, and also test this codes in 1.8 for allowed_buildroot_push, we removing these codes it will keep the previous value when I put without a value.
But adding these codes it will set default without a value in the parameters. so removing the codes in 1.11 get the same results.
Here from my view it's ok to remove these codes, not sure if we should ask QE to test this, hope there is no issue in ct's scripts :)

xiangge replied:
lol, you and me did the same test

xiangge commented on pdc/apps/auth/__init__.py:
```diff
@@ -16,7 +16,7 @@
 
-    MIDDLEWARE_CLASSES = (
+    MIDDLEWARE = (
         ...
```
Do we need to change the MIDDLEWARE_CLASSES to MIDDLEWARE in settings.py?

hluk replied:
Nice find. Fixed.

xiangge commented on verify-migrations.sh:
```diff
@@ -1,7 +1,2 @@
-if python manage.py makemigrations --dry-run --exit; then
-    exit 1
-fi
+exec python manage.py makemigrations --check
```
Do we need use --dry-run --check? Seems --check is same as --exit?

hluk replied:
Seems that `--check` doesn't change DB either (same as `--dry-run`).

hluk replied:
Sorry, of course the commands don't change DB, but both the new command and the old one doesn't create any migration files.

xiangge commented:
The patch is almost ok for me besides some tiny concerns, please see my comments, and thx for adding me as a reviewer :)

simozhan commented:
It seems docs/source/development.rst need to update as well.

hluk commented:
> It seems docs/source/development.rst need to update as well.

Updated. I also bumped the versions in `pdc.spec`.

> Maybe we can add --parallel for the test which can significantly reduce the test time, example in Makefile: "python manage.py test --settings pdc.settings_test pdc contrib"

Changed to:
```bash
python manage.py test --settings pdc.settings_test --parallel -- pdc contrib
```

Not sure if I can change it for the coverage run (also for Travis CI).

lubomir commented:
Travis CI configuration is in https://github.com/product-definition-center/product-definition-center/blob/master/.travis.yml, but I'm not sure if it needs to be changed. It should have [two CPUs available](https://docs.travis-ci.com/user/reference/overview/#Virtualisation-Environment-vs-Operating-System), but I have no idea how it would affect measuring the code coverage.

hluk commented:
I've rebased the changes to the latest commit in master branch.

Any last calls before I merge this?

lubomir commented:
:+1: Go for it.

hluk commented:
There you go! Merged!

xiangge commented:
OK for me.
